### PR TITLE
Demo run profile

### PR DESCRIPTION
**Pull request by wujimin**
JAV-92 create a pom profile to make demos can run by java -jar xxx.jar

mvn install -Pdemo-run-release
  will create demo-bin directory in the root of java-chassis

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -75,3 +75,4 @@ __pycache__/
 
 # SVN metadata
 .svn/
+demo-bin
```
Comment (WillemJiang) [inline .gitignore]: Can we  put the demo-bin into the target directory which is ignored by the git.

File: demo/pom.xml [hunk 2]
```diff
@@ -165,4 +165,74 @@
...
+	
+	<profiles>
+		<profile>
+			<id>demo-run-release</id>
+			<build>
+				<plugins>
+					<plugin>
+						<groupId>org.commonjava.maven.plugins</groupId>
+						<artifactId>directory-maven-plugin</artifactId>
+						<version>0.1</version>
```
Comment (seanyinx) [inline demo/pom.xml]: the version numbers of the 3 plugins can be inherited from parent

File: .gitignore (hunk 1 above)
Comment (wujimin) [inline .gitignore]: changed to root/demo/target

File: demo/pom.xml (hunk 2 above)
Comment (wujimin) [inline demo/pom.xml]: removed these versions

**Pull request merged**